_infra/helm/locust/locustfiles/locustfile.py

```diff
@@ -529,24 +529,34 @@ def perform_requests(self):
             grouping = request.get("grouping")
             expected_response_text = request.get("expected_response_text")
             expected_response_status = request.get("response_status", 200)
+            request_url = request["url"] if "url" in request else None
+            harvest_dict = {}
 
-            if self.response and "harvest_url" in request:
+            if self.response and "harvest" in request:
                 soup = BeautifulSoup(self.response.text, "html.parser")
-
-                for link in soup.find_all(id=request["harvest_url"]["id"]):
-                    if request["harvest_url"]["link_text"] in link.get_text():
-                        request_url = link.get("href")
-                        break
-                    logger.error(f"Unable to harvest url {request['harvest_url']}")
-                    self.interrupt()
-            else:
-                request_url = request["url"]
+                harvest_details = request["harvest"]
+
+                if harvest_details["type"] == "url":
+                    for link in soup.find_all(id=request["harvest"]["id"]):
+                        if request["harvest"]["link_text"] in link.get_text():
+                            request_url = link.get("href")
+                            break
+                        logger.error(f"Unable to harvest url {request['harvest']}")
+                        self.interrupt()
+
+                if harvest_details["type"] == "id":
+                    for id_name in harvest_details["ids"]:
+                        input_name = soup.find("input", attrs={"name": id_name})
```

Comment:
This doesn't quite tie together, we are searching via name not id, so the identifier in the JSON should match. Realistically we shouldn't be using name, but I can see that the code in frontstage doesn't have an id. A card should be raised to change that and then this to id

Reply:
I'm happy tto change it back to value for the moment and I'll raise a separate card for tech session to address this.